invenio_circulation/errors.py

```diff
@@ -8,50 +8,142 @@
 
 """Circulation exceptions."""
 
+import json
 
-class CirculationException(Exception):
-    """Exceptions raised by circulation module."""
+from flask import current_app
+from invenio_rest.errors import RESTException
 
-    def __init__(self, msg):
-        """Constructor."""
-        self.msg = msg
 
+class CirculationException(RESTException):
+    """Base Exception for circulation module, inherit, don't raise."""
 
-class InvalidState(CirculationException):
-    """State not found in circulation configuration."""
+    code = 400
 
+    @property
+    def name(self):
+        """The status name."""
+        return type(self).__name__
 
-class ItemNotAvailable(CirculationException):
-    """Exception raised from action on unavailable item."""
+    def __init__(self, **kwargs):
+        """Initialize exception."""
+        super(CirculationException, self).__init__(**kwargs)
 
+    def get_response(self, environ=None, **kwargs):
+        """Intercept response to add info and log the error."""
+        resp = super(CirculationException, self).get_response(environ=environ)
+        data = json.loads(resp.data.decode('utf-8'))
+        data["error_module"] = "Circulation"
+        data["error_class"] = self.name
+        resp.data = json.dumps(data)
+        current_app.logger.exception(self)
```

Comment:
I think we should not log it, because otherwise we will always have a Sentry exception logged.
For unhandled exception, it should happen, but not for handled...